### ISSUE TITLE
Show user avatars in User list

### ISSUE THREAD

**Issue by pixelrazor** [nitnelave/lldap#492]
Can leverage the avatar component from #407

**Comment by nitnelave** [comment]
Some things to think about:
 - Caching (to avoid re-fetching all the users' avatars every time)
 - Keyed events in the avatar agent, with the username? We don't want to re-fetch all the users' avatars anytime a single one changes.

**Comment by pixelrazor** [comment]
With rrespect those concerns, I'm leaning towards the ffollowing solution:

- return avatar in GetUserDetail
- Storing the user detail list in global state (whether  that be Context or we end up using a lib like yewdux)

**Comment by nitnelave** [comment]
We could store the avatar in the agent: actually, you could move the request from the avatar display to the agent, and the agent would call you back with the avatar, either from cache or from asking the server.

**Comment by pixelrazor** [comment]
Well this is going to be multiple avatars, multiple users. I have an initial impl i'll throw up as a draft that loads them after page/table load - while t his is the simpler approach with less changes, i also like it because the initial table and isn't blocked on waiting for all the avatars (a potentially big payload if it's included in the main get of all the users). Anytime you navigate away from the users page, it will refresh when you go back to it anyways, so i don't see a big benefit from adding the complexity here personally.

If we can really justify the larger changes, i'd probably nuke the agent (the Context agents were actually removed, and context/hooks are the preferred way now) and write a component to handle that state and provide it through context or from some third party state lib

**Comment by nitnelave** [comment]
Right, but the agent can have a hash map of the avatars, queried by username.
I was thinking that the user table would just instantiate the avatar components, which would request the avatar from the agent (maybe with some queueing/grouping if it makes sense? In a future update). The Avatar component could then use the suspense API from yew to render a placeholder, and then get called back by the agent when the image is received.
That's also not blocking the user table.

**Comment by pixelrazor** [comment]
I think i'm convinced enough. Also of note, agents are now called Workers in yew-agent 0.2.0
